# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/LoadedScript.h
# Commit: a50f60a612d3
# Full Hash: a50f60a612d31de79369e59933242ee10a5d810f
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 09:08:22
# Regressor Bug: 1742437
# File Overlap Count: 1
# Description:
#   Bug 1742437 - move mElement from ScriptFetchOptions to DOMScriptLoadContext; r=jonco,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D136742
# ==============================================================================

diff -r 0ac6d9d26e2b -r a50f60a612d3 dom/script/LoadedScript.h
--- a/dom/script/LoadedScript.h	Thu Feb 17 15:57:48 2022 +0000
+++ b/dom/script/LoadedScript.h	Thu Feb 17 15:57:48 2022 +0000
@@ -17,6 +17,7 @@
 namespace mozilla {
 namespace dom {
 
+class Element;
 class ScriptLoader;
 
 void HostAddRefTopLevelScript(const JS::Value& aPrivate);
@@ -30,10 +31,11 @@
   ScriptKind mKind;
   RefPtr<ScriptFetchOptions> mFetchOptions;
   nsCOMPtr<nsIURI> mBaseURL;
+  nsCOMPtr<Element> mElement;
 
  protected:
   LoadedScript(ScriptKind aKind, ScriptFetchOptions* aFetchOptions,
-               nsIURI* aBaseURL);
+               nsIURI* aBaseURL, Element* aElement);
 
   virtual ~LoadedScript();
 
@@ -48,7 +50,12 @@
   inline ModuleScript* AsModuleScript();
   inline EventScript* AsEventScript();
 
+  // Used to propagate Fetch Options to child modules
   ScriptFetchOptions* GetFetchOptions() const { return mFetchOptions; }
+
+  // Used by the Debugger to get the associated Script Element
+  Element* GetScriptElement() const { return mElement; }
+
   nsIURI* BaseURL() const { return mBaseURL; }
 
   void AssociateWithScript(JSScript* aScript);
@@ -58,14 +65,16 @@
   ~ClassicScript() = default;
 
  public:
-  ClassicScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL);
+  ClassicScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
+                Element* aElement);
 };
 
 class EventScript final : public LoadedScript {
   ~EventScript() = default;
 
  public:
-  EventScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL);
+  EventScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
+              Element* aElement);
 };
 
 // A single module script. May be used to satisfy multiple load requests.
@@ -83,7 +92,8 @@
   NS_DECL_CYCLE_COLLECTION_SCRIPT_HOLDER_CLASS_INHERITED(ModuleScript,
                                                          LoadedScript)
 
-  ModuleScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL);
+  ModuleScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
+               Element* aElement);
 
   void SetModuleRecord(JS::Handle<JSObject*> aModuleRecord);
   void SetParseError(const JS::Value& aError);